reference/FungibleToken.scilla

```diff
@@ -0,0 +1,429 @@
+  allowances_map[_sender][spender] := amount;
+  e = {_eventname : "ApproveSuccess"; sender : _sender; spender: spender; amount : amount};
+  event e
+end
+
+(* Transitions *)
+
+(* @dev: Moves amount tokens from _sender to the recipient. Used by token_owner *)
+(* @param recipient:  Address of the recipient whose balance is increased.      *)
+(* @param amount:     Amount of tokens to be sent.                              *)
+transition Send(to: ByStr20, amount: Uint128)
+  ProcedureMove _sender to amount
+end
+
+(* @dev: Moves amount tokens from token_owner to recipient. _sender must be an operator of token_owner. *)
+(* @param from:        Address of the token_owner whose balance is decreased.                           *)
+(* @param to:          Address of the recipient whose balance is increased.                             *)
+(* @param amount:      Amount of tokens to be sent.                                                     *)
+transition OperatorSend(from: ByStr20, to: ByStr20, amount: Uint128)
+  get_operator <- operators_map[from][_sender];
+  is_operator_approved = 
+    match get_operator with
+    | Some value => value
+    | None => False
+    end;
+  is_default_operator = is_default_operator f_eq _sender default_operators;
+  is_revoked_operator <- exists revoked_default_operators[from][_sender];
+  is_default_operator_approved = let is_not_revoked_operator = 
+    negb is_revoked_operator in andb is_not_revoked_operator is_default_operator;
+  is_approved = orb is_operator_approved is_default_operator_approved;
+  match is_approved with
+    | False =>
+      err = CodeNotAuthorized;
+      MakeError err
+    | True =>
+      ProcedureMove from to amount
+   end
+end
+
+(* @dev: Move a given amount of tokens from one address to another using the allowance mechanism. The caller must be an approved_spender. *)
+(* @param spender: Address of the spender who is tied to the token_owner's assets and have spending allowances allocated by token_owner.  *)
+(* @param from:    Address of the token_owner whose balance is deccreased.                                                                *)
+(* @param to:      Address of the recipient whose balance is increased.                                                                   *)
+(* @param amount:  Amount of tokens to be transferred.                                                                                    *)
+transition TransferFrom(from: ByStr20, to: ByStr20, amount: Uint128)
+  get_bal <- balances_map[from];
+  match get_bal with
+  | None => 
+    err = CodeNotFound;
+    MakeError err
+  | Some bal =>
+    get_spender_allowed <- allowances_map[from][_sender];
+    match get_spender_allowed with
+    | None =>
+      err = CodeNotAuthorized;
+      MakeError err
+    | Some allowed =>
+      min = min_int bal allowed;
+      can_do = uint128_le amount min;
+      match can_do with
+      | True =>
+        new_from_bal = builtin sub bal amount;
```

Comment:
Can you not call `ProcedureMove` here?

Reply:
Due to this `min = min_int bal allowed;` requirement, we cannot call `ProcedureMove` as it takes only `balance` into account.